src/pybind/mgr/cephadm/schedule.py

```diff
@@ -49,6 +49,52 @@ def renumber_ports(self, n: int) -> 'DaemonPlacement':
             self.rank_generation,
         )
 
+    def assign_rank(self, rank: int, gen: int) -> 'DaemonPlacement':
+        return DaemonPlacement(
+            self.daemon_type,
+            self.hostname,
+            self.network,
+            self.name,
+            self.ip,
+            self.ports,
+            rank,
+            gen,
+        )
```

Comment:
Is this the best place to put that code. Would be great to have the rank assignment in the `NFSService` class.